utils/funcexp/func_math.cpp

```diff
@@ -1316,6 +1514,21 @@ double Func_tan::getDoubleVal(Row& row,
         }
         break;
 
+        case execplan::CalpontSystemCatalog::LONGDOUBLE:
+        {
+            // null value is indicated by NaN
+            long double value = parm[0]->data()->getLongDoubleVal(row, isNull);
+
+            if (isNull || value <= 0.0)
```

Comment:
This may be a typo. I believe it should match line 1507. We could take the tangent of a negative

Reply:
Also correct